### PR TITLE
Consume availability interface for cores

### PR DESCRIPTION
**Pull request by lxwinspur**
In the current state, the logic is bmcweb for displaying cores is:
It checks the Present property on core dbus object to update
the [Status][State] as Enabled/Absent.

In addition to Present property, we should also use the Available
property , if the Available is false, then set the [Status][State]
should be set as "UnavailableOffline".

Refer: https://github.com/ibm-openbmc/dev/issues/3481

Signed-off-by: George Liu <liuxiwei@inspur.com>

### CODE REVIEW

File: redfish-core/lib/processor.hpp [hunk 1]
```diff
@@ -1052,15 +1071,24 @@ inline void
...
                 }
                 else
                 {
-                    if (!functional)
+                    if (present == false)
                     {
-                        aResp->res.jsonValue["Status"]["Health"] = "Critical";
+                        aResp->res.jsonValue["Status"]["State"] = "Absent";
+                    }
+                    else
```
Comment (gtmills) [inline redfish-core/lib/processor.hpp:1085]: Why not  
if
else if 
else if

**Comment by gtmills** [comment]
Looks fine, @lxwinspur can you push upstream and use 
if
else if
else if

instead?

File: redfish-core/lib/processor.hpp (hunk 1 above)
Comment (lxwinspur) [reply redfish-core/lib/processor.hpp]: Ack

File: redfish-core/lib/processor.hpp [hunk 2]
```diff
@@ -1052,15 +1071,24 @@ inline void
...
+                    {
+                        if (!functional)
```
[the remark this answers is not in the document]
Comment (lxwinspur) [reply redfish-core/lib/processor.hpp]: Done

**Comment by lxwinspur** [comment]
> Looks fine, @lxwinspur can you push upstream and use if else if else if
> 
> instead?

I have updated this issue.

BTW, Considering that this patch has not been reviewed, do we still need upstream?
https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/38570

**Review by gtmills — APPROVED** [review]

**Comment by gtmills** [comment]
jenkins run tests please

**Pull request merged**